### ISSUE TITLE
CDP "Network.setUserAgentOverride" not work in ServiceWorker

### ISSUE THREAD

**Issue by l0ca1** [puppeteer/puppeteer#8867]
### Bug description

CDP "Network.setUserAgentOverride" not work in ServiceWorker.

Steps to reproduce the problem:
```const puppeteer = require('puppeteer');
(async () => {
  const browser = await puppeteer.launch({
    args: ["--enable-experimental-web-platform-features"],
    headless: false,
  });
  browser.on("targetcreated", async (target) => {
    if (target.type() === "service_worker") {
      const client = await target.createCDPSession();
      if (!client) return;
      console.log("service worker set useragent");
      await client.send("Network.setUserAgentOverride", {
        "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36",
        "platform": "MacIntel",
        "userAgentMetadata": {
          "brands": [
            {
              "brand": "Chromium",
              "version": "100"
            },
            {
              "brand": "Google Chrome",
              "version": "100"
            },
            {
              "brand": ";Not A Brand",
              "version": "96"
            }
          ],
          "fullVersion": "100.0.4896.127",
          "platform": "Mac OS X",
          "platformVersion": "10_15_7",
          "architecture": "x86",
          "model": "",
          "mobile": false
        }
      });
      const worker = await target.worker();
      if (!worker) return;
      const data = await worker.evaluate(() => {
        if ('userAgentData' in navigator) {
          return navigator.userAgentData.brands;
        }
        return undefined;
      }).catch(e => {
        console.log(e.message);
        return undefined;
      });
      console.log("userAgentData: " + JSON.stringify(data));
    }
  });
  const page = await browser.newPage();
  await page.goto('https://abrahamjuliot.github.io/creepjs/');
  // await browser.close();
})();
```
I expect to get log:
`userAgentData: [{ "brand": "Chromium", "version": "100" }, { "brand": "Google Chrome", "version": "100" }, { "brand": ";Not A Brand", "version": "96" }]`

but i get result log:
`userAgentData: [{"brand":"Chromium","version":"105"},{"brand":"Not)A;Brand","version":"8"}]`

userAgent have the same results.

### Puppeteer version

v17.0.0

### Node.js version

v14.20.0

### npm version

v6.14.15

### What operating system are you seeing the problem on?

macOS

**Comment by OrKoN** [comment]
I believe it's a known issue and basically it's https://github.com/puppeteer/puppeteer/issues/3667 What happens is that when the client sends Network.setUserAgentOverride it's too late as the service worker already initiated network requests.

**Comment by OrKoN** [comment]
Actually, I probably misread the description: have you tried to use https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setUserAgentOverride as Network.setUserAgentOverride might be only affecting network requests.

**Comment by l0ca1** [comment]
> Actually, I probably misread the description: have you tried to use https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setUserAgentOverride as Network.setUserAgentOverride might be only affecting network requests.

yes,i have tried use `Emulation.setUserAgentOverride`,the result is the same：
`userAgentData: [{"brand":"Chromium","version":"105"},{"brand":"Not)A;Brand","version":"8"}]`

**Comment by OrKoN** [comment]
Filed https://crbug.com/1358491

**Comment by mihneamanolache** [comment]
Any luck fixing this issue?

**Comment by kaliiiiiiiiii** [comment]
I think there might be a way as following:
use `` to execute a script on a new page which
- [ ] apply a `Proxy` to `["Worker", "SharedWorker", "navigator.serviceWorker.register"]` which detects, when a __worker gets initialized__.
- [ ] bevore a worker gets initialized, inject some code into into it, which modifies `"navigator.userAgentData", "navigator.plaform", "navigator.userAgent"`
- [ ] Fix issues with detection
      - on the proxies
      - on the modified objects in the `Worker`
      - [article](https://adtechmadness.wordpress.com/2019/03/23/javascript-tampering-detection-and-stealth/) might help with that, but still doesn't provide a full solution :)
      
Unfortanely, __I don't have enough expirience__ in javascript to realize those things, maybe @OrKoN @l0ca1 @mihneamanolache @jeresig or @benatkin got some ideas?

**Comment by divinity76** [comment]
> Actually, I probably misread the description: have you tried to use https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setUserAgentOverride as Network.setUserAgentOverride might be only affecting network requests.

fwiw `Network.setUserAgentOverride` also affects javascript's `navigator.language` and `navigator.languages` for javascript-based language detection.

**Comment by tzbo** [comment]
Any update?

**Comment by pgamaster** [comment]
I can confirm the bug

**Comment by LeninZapata** [comment]
For that you need a modified chromium at the assembly level, something like using a disassembler-type program. And put the variables above all "platform" to be able to edit this. I think it's the most complicated thing of all about bots.
